homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+
+import Element from "./components/Element/Element";
+import Interval from "./components/Interval/Interval";
+import Controls from "./components/Controls/Controls";
+
+import {generateRandomArray, checkIfSorted, interval} from "./helpers";
+import {DEFAULT_ARRAY_SIZE, DEFAULT_INTERVAL} from "./constants";
+import './App.css';
+
+enum Status {
+    Idle = 'idle',
+    Pause = 'pause',
+    Sort = 'sort',
+    Solved = 'solved'
+}
+
+type State = {
+    data: number[],
+    status: Status,
+    startIndex: number,
+    sortInterval: number,
+    intervalInput: number,
+    sortByStep: boolean,
+}
+
+class App extends React.Component<any, State> {
+
+    state: State = {
+        data: generateRandomArray(DEFAULT_ARRAY_SIZE),
+        status: Status.Idle,
+        startIndex: 0,
+        sortInterval: DEFAULT_INTERVAL,
+        intervalInput: DEFAULT_INTERVAL,
+        sortByStep: false,
+    }
+
+    sort = async (): Promise<void> => {
```

Comment:
Better to keep a business logic separate from the component. E.g., create a pure function and move it to another file (e.g., under `utils` folder).